VSRAD.DebugServer/SharedUtils/FileMetadata.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Serialization;
+using System.Text;
+
+namespace VSRAD.DebugServer.SharedUtils
+{
+
+    public class FileMetadata// : IEquatable<FileMetadata>
+    {
+        /// <summary>
+        /// Paths use / as the directory separator. Paths ending with the slash are interpreted as directories:
+        /// this matches the behavior of zip archives used in <see cref="IPC.Commands.GetFilesCommand"/> and <see cref="IPC.Commands.PutDirectoryCommand"/>.
+        /// </summary>
+        public string relativePath_ { get; set; }
+
+        public DateTime lastWriteTimeUtc_ { get; set; }
+
+        public bool isDirectory_ { get; set; }
+
+        public FileMetadata() { }
+
+        public FileMetadata(string relativePath, DateTime lastWriteTimeUtc, bool isDirectory)
+        {
+            relativePath_ = relativePath;
+            lastWriteTimeUtc_ = lastWriteTimeUtc;
+            isDirectory_ = isDirectory;
+        }
+
+        public String ToString()
+        {
+            var props = this.GetType().GetProperties();
+            var sb = new StringBuilder();
+            foreach (var p in props)
+            {
+                sb.AppendLine(p.Name + ": " + p.GetValue(this, null));
+            }
+            return sb.ToString();
+        }
```

Comment:
Do we really need to use reflection to iterate over 3 properties?

Also, readability can be improved. I would suggest maybe something more like

```c#
public override string ToString() => string.Join(Environment.NewLine, new string[]
{
    "Metadata = [",
    $"\tRelative Path       : {relativePath_}",
    $"\tLast Write Time UTC : {lastWriteTimeUtc_}",
    $"\tIs Directory        : {isDirectory_}",
    "]"
});
```